src/promise-katas-2.js

```diff
@@ -25,18 +25,27 @@ fetch("joke", "question").then(<insert your callback function>)
 // 1 Create a function that uses the fetch function to make a request to the "food" URL and returns
 // the data - expected return value "Cheese" of type String
 
-const food = () => {};
+const food = () => {
+  return fetch("food").then((food) => { return food.data })
+};
 
 // 2 Create a function that uses the fetch function to make a request to the "cats" URL and returns
 // a list of cats in alphabetical order - expected return value ["Bandit", "Berry", "Puss in boots", "Smokey"] of type Array
 
-const cat = () => {};
+const cat = () => {
+  return fetch("cats").then((cat) => { return cat.data.cats.sort() });
+};
 
 // 3 Create a function that uses the fetch function to make a request to the "dogs" URL and returns
 // the naughtiest dog - expected return value {name: "Mutley", naughty: 10} of type Object
 
-const dog = () => {};
-
+const dog = () => {
+  return fetch("dogs").then((dog) => {
+    const naughtyDog = dog.data.dogs.reduce((a, b) => a.naughty > b.naughty ? a : b);
+    return naughtyDog;
```

Comment:
No need to create the variable here, can just return the value of the reduce function:
```
    return dog.data.dogs.reduce((a, b) => a.naughty > b.naughty ? a : b);
```